File: javasrc/src/ice/ui/stat/IceStat.java
package ice.ui.stat;

import arc.Core;
import arc.struct.Seq;
import mindustry.world.meta.Stat;

public class IceStat extends Stat {

    public static Stat nutrientConcentration;
    public static Seq<Stat> allStat = new Seq<>();

    static {
        nutrientConcentration = new IceStat("nutrientConcentration", "[red]", "{shake}");
    }

    public String color;
    public String defaultToken;

    public IceStat(String name) {
        this(name, "[lightgray]");
    }

    public IceStat(String name, String color) {
        this(name, color, "");
        this.color = color;
    }

    public IceStat(String name, String color, String defaultToken) {
        super(name);
        allStat.add(this);
        this.color = color;
        this.defaultToken = defaultToken;
    }

    @Override
    public String localized() {
        return Core.bundle.get("stat." + name);
    }
}

File: src/ice/world/meta/IceStat.kt
package ice.world.meta

import mindustry.world.meta.Stat
import mindustry.world.meta.StatCat
import universecore.ui.bundle.Localizable

class IceStat(name: String, category: StatCat = StatCat.general) :Stat(name, category), Localizable {
  override var localizedName: String = ""
  override var description: String = ""
  override var details: String = ""

  override fun localized() = localizedName
  override fun toString() = localizedName
}